dvc/stage.py

```diff
@@ -105,11 +105,15 @@ def link(self, checkout=False):
             src = self.path
             link = self.cache
         else:
-            raise OutputNoCacheError(self.path)
+            self.project.logger.debug('Output \'{}\' is not in cache'.format(self.path))
+            if os.path.exists(self.path):
+                os.unlink(self.path)
```

Comment:
link() should not return False nor True. If it is called for an output that doesn't have cache to perform checkout, then it should raise an exception, as it means that user should obtain cache by pulling data from the cloud. It definitely should not remove self.path if cache is not found, as there might be some valuable data that is not yet cached.

Could you please elaborate on the issue you are trying to solve?